design/open/epsilon-unity/epsilon-unity-exporter.md

```diff
@@ -0,0 +1,195 @@
+  could be post- hooks, but pre- hooks will probably also be usable.
+
+### Writer
+
+The writer writes a Delta Lake file as objects to lakeFS.  It can use
+lakeFSFS or the S3 gateway, of course, along with appropriate lakeFS
+credentials.  All features of lakeFS can be used to manage versions of the
+Delta Lake file.
+
+They then trigger an Epsilon exporter run.  It translates Delta Lake
+metadata objects to be available to Unity at a known provider blockstore
+path.  Unity is not involved in this path, beyond defining an external table
+at a given (known) location in a given catalog.
+
+![lakeFS write path: Spark writes DeltaLake files using lakeFS for metadata and either lakeFS or S3 directly for data](./lakeFS-write-path.png)
+
+### Reader
+
+Readers read a Delta Lake file as a Unity external table on a catalog.  We
+_recommend_ using a schema named for the exported ref.  This external table
+is based on the translation performed by the Epsilon exporter.  lakeFS is
+not involved in this path.
+
+![Unity read path: Unity reads DeltaLake files directly as an external table in a catalog on S3](./Unity-read-path.png)
+
+## Epsilon exporter flow
+
+The Epsilon exporter is a component external to the lakeFS daemon and also
+to Unity.
+
+It _reads_ Delta Lake metadata the lakeFS repository at a given ref and
+path.  The Delta Lake repository describes the [Delta Transaction Log
+Protocol][deltalake-format] used for this Delta Lake metadata.  Some fields
+hold _path references_ to specific Delta Lake data objects.[^1]
+
+The input Delta Lake file is at some location
+`lakefs://REPO/REF/input/path/`.  That URI can hold data files (Parquets).
+It also holds metadata files under a prefix `_delta_log`.  The output Delta
+Lake file will be at some location `s3://BUCKET/output/path/`.  It holds
+only metadata, so it holds only objects in prefix
+`s3://BUCKET/output/path/_delta_log/`.
+
+Epsilon _transforms_ the metadata of the input file:
+
+* In the first version we shall drop CRC files.  A later version will
+  restore these.  (These files are 
+* For every path reference in the metadata, it creates a _translated path
+  reference_.  The translated path reference is always an absolute on the
+  provider blockstore:
+  - An _absolute_ path reference `lakefs://REPO/REF/...`, is translated to
+    its matching physical path on the provider blockstore by querying lakeFS.
+  - A _relative_ path reference `.../...` is expanded into an absolute path
+    reference under the root directory, then that absolute path is
+    translated to its matching physical path on the provider blockstore by
+    querying lakeFS.
+  - Delta Lake "DV files" may have an additional "base-85" path encoding.
+    Such a path has its base-85 decoded into a UUID, that UUID is then
+    expanded to a relative path, which is finally translated into its
+    matching physical path on the provider blockstore by querying lakeFS.
```

Comment:
I wonder if we can configure Delta to write only relative paths and then just copy without translation.